src/epcpm/sunspectointerface.py

```diff
@@ -0,0 +1,2227 @@
+                        ]
+                    )
+                    register_index += 1
+                    addr_index += 1
+        c_lines.extend(
+            [
+                "};\n\n",
+            ]
+        )
+
+        c_lines.extend(
+            [
+                f"ModbusReg * const sunspec{self.sunspec_id.value}AddrRegMap[{total_addresses}] =\n",
+                "{\n",
+            ]
+        )
+        # Account for 'SunS' length.
+        register_index = epcpm.pm_helper.SUNS_LENGTH
+        point_index = 0
+        for block_points in model_points:
+            for point in block_points:
+                if point.type == "SunspecModelHeader":
+                    register_index += point.size
+                else:
+                    for point_size_index in range(point.size):
+                        c_lines.extend(
+                            [
+                                f"    [{register_index}] = &sunspec{self.sunspec_id.value}Regs[{point_index}],\n"
+                            ]
+                        )
+                        point_index += 1
+                        register_index += 1
+        c_lines.extend(
+            [
+                "};\n\n",
+            ]
+        )
+
+        c_lines.extend(
+            [
+                f"SunspecModel sunspec{self.sunspec_id.value}ModelList[] =\n",
+                "{\n",
+            ]
+        )
+        for model_id in model_list:
+            c_lines.extend(
+                [
+                    f"    {{.num = {model_id}, .data = &sunspec{self.sunspec_id.value}Interface.model{model_id}, .init = &sunspec{self.sunspec_id.value}Model{model_id}Init}},\n"
+                ]
+            )
+        c_lines.extend(
+            [
+                "};\n\n",
+            ]
+        )
+
+        # Output the overall .c file.
+        with self.c_path.open("w", newline="\n") as f:
+            for c_line in c_lines:
+                f.write(c_line)
+
+        # Output the overall .h file.
+        with self.h_path.open("w", newline="\n") as f:
+            for h_line in h_lines:
+                f.write(h_line)
+
+
+@builders(epcpm.sunspecmodel.Model)
+@attr.s
+class Model:
+    """Interface generator for the SunSpec Model class."""
+
+    wrapped = attr.ib()
+    parameter_uuid_finder = attr.ib()
+    sunspec_id = attr.ib()
+    skip_sunspec = attr.ib()
+    c_path = attr.ib(default=None)
+    h_path = attr.ib(default=None)
+
+    def gen(self) -> typing.Tuple[typing.List[str], int, typing.List[OutputPoint]]:
+        """
+        Interface generator for the SunSpec Model class.
+
+        Returns:
+            list of strings, int, list of OutputPoint: output .h lines, model ID, list of model OutputPoint's
+        """
+        points = []
+        h_enum_lines = []
+        h_enum_default_enumerator_dict = dict()
+        model_types = ["Header", "Fixed Block", "Repeating Block"]
+        zipped = zip(enumerate(self.wrapped.children), model_types)
```

Comment:
zipped isn't that intuitive of a variable name. Perhaps zipped_children_models or something?

Reply:
Renamed in a few places. Prior author's name choice. b265d71d5b1682a497fa7f7520dbbd5599753550